### PR TITLE
Assignment 1

### PR DESCRIPTION
**Pull request by Carvas91**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
I am adding Lab 1, 2, 3, and Assignment 1, which include training deep learning models, experimenting with image classification, and improving model performance using regularization and augmentation techniques.
## What did you learn from the changes you have made?
I had the opportunity to use image augmentation libraries to improve generalization. I also learned how to reduce overfitting using L2 regularization and how to analyze model performance using training curves and classification probabilities.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
Yes, I considered using a deeper architecture (ResNet-style residual connections) to improve feature extraction. 

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
Yes, training times were a challenge. Even after 50 epochs, validation accuracy was still improving, indicating potential for further improvement. However, training was time-consuming, and I had to submit the assignment before reaching optimal performance.

## How were these changes tested?
Yes, training times were a challenge. Even after 50 epochs, validation accuracy was still improving, indicating potential for further improvement. However, training was time-consuming, and I had to submit the assignment before reaching optimal performance.

## A reference to a related issue in your repository (if applicable)
NA

## Checklist
- ✅ I can confirm that my changes are working as intended

### CODE REVIEW

**Review by x-rojas-io — APPROVED** [review]
Hi Carlos!

Thank you for submitting your DL assignments.

Your A1 and Lab 1-3 are now Complete.

Thanks,
Nestor [LS]

**Comment by Carvas91** [comment]
Hi @nestorojas thank you very much for your feedback!
